superset/dashboards/schemas.py

```diff
@@ -198,6 +199,15 @@ class DashboardGetResponseSchema(Schema):
     changed_on_humanized = fields.String(data_key="changed_on_delta_humanized")
     is_managed_externally = fields.Boolean(allow_none=True, dump_default=False)
 
+    # pylint: disable=unused-argument
+    @post_dump()
+    def post_dump(self, serialized: dict[str, Any], **kwargs: Any) -> dict[str, Any]:
+        if security_manager.is_guest_user():
+            serialized["owners"] = []
```

Comment:
I would prefer to fully remove the property `owners`, as replacing it with an empty list is IMO conveying incorrect info

Reply:
@villebro I agree. The only caveat would be if the owners field is required.

Reply:
@john-bodley ideally I think we should relax that requirement if we want to make this data conditional on the permissions of the logged in user. Based on the proposed changes, I think it should be as follows:
- regular user: all relevant owner info is conveyed (current behavior)
- guest token: no owner property is present in the response

Reply:
@villebro I initially opted to keep the owners property in the payload because the dashboard frontend expects it to always be there, and fails to render if it's not.

I found a way to exclude it from the response payload and have the client tack the property on after the dashboard is requested, which allows the embedded dashboard to render successfully.